test/src/main/java/org/springframework/security/test/context/support/WithMockOidcUserSecurityContextFactory.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.test.context.support;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.user.DefaultOidcUser;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.util.StringUtils;
+
+import java.time.Instant;
+import java.util.*;
+
+/**
+ * A {@link WithMockOidcUserSecurityContextFactory} that works with {@link WithMockOidcUser}.
+ * <p>
+ * Initializes the Spring Security Context with a OAuth2AuthenticationToken instance, which comes with
+ * an encoded oidc token with some default claims but without header and signature.
+ *
+ * @author Nena Raab
+ * @see WithMockOidcUser
+ * @since 5.3
+ */
+final class WithMockOidcUserSecurityContextFactory implements
+		WithSecurityContextFactory<WithMockOidcUser> {
+
+	public SecurityContext createSecurityContext(WithMockOidcUser withUser) {
+		String userId = StringUtils.hasLength(withUser.name()) ? withUser
+				.name() : withUser.value();
+		if (userId == null) {
+			throw new IllegalArgumentException(withUser
+					+ " cannot have null userId on both userId and value properties");
+		}
+
+		List<GrantedAuthority> grantedAuthorities = new ArrayList<>();
+		for (String authority : withUser.authorities()) {
+			grantedAuthorities.add(new SimpleGrantedAuthority(authority));
```

Comment:
To align this with the `OidcUser` that `OidcUserService` creates, I think [an `OidcUserAuthority` should also be added](https://github.com/spring-projects/spring-security/blob/master/test/src/main/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestPostProcessors.java#L1553).

Reply:
The only thing it does now: it adds another `GrantedAuthority` with "ROLE_USER", not sure why required...